### ISSUE TITLE
v2 Settings Interface

### ISSUE THREAD

**Issue by spindrjr** [zininzinin/AutoTrimps#24]
@Ishakaru has proposed a much-needed overhaul of the user settings interface. Discuss/suggest features here.

**Comment by WG-SpaceCoder** [comment]
I was imagining a whole new interface, but I would be okay with changing all the checkboxes to buttons and having a whole new screen for each 'section' of automation. Like maps, jobs, equip each have their own popup menu or something like that? Also having a 'simple' mode would be pretty cool with all of our defaults configured.

**Comment by spindrjr** [comment]
Personally I hate the idea of having to use multiple screens for multiple sections. Much rather have one large screen with everything. Either a flyup from the bottom or a popup both seem fine to me.

**Comment by Ishakaru** [comment]
The overall goal is to have the automation feel like it's part of the game rather than something just tacked on. On the back end(code) I want the addition of new features to be easy(ier) to include. 

My proposed change is to implement a wrapper to contain all the options. How it's presented to the player to be similar to how the current settings window pops up. In the end, I want it to function exactly like the settings menu for all "checkbox" options. For value ranges I want the custom buy input to be used. So no text boxes just sitting there. 

On the back end(code) I want to use several arrays to handle the creation, and data of the options. I haven't figured out a good way to procedurally place the options, so my first pass will involve using several different arrays that get used by the setup function. (Death to .innerHTML+='string for days' manipulation)

**Comment by WG-SpaceCoder** [comment]
Dude that sounds awesome. Let me know if you get a prototype going or if you need any support on this! How does it sound having another button at the bottom that pops up like setting or even view perks. Having the new popup have tabbed sections would be nice too. Thoughts?

**Comment by Ishakaru** [comment]
I want it to work exactly like settings. Not sure if I want to mess with tabs... Lets leave that for a modification down the road.

I think I'll be done by Sunday evening at the latest. I'm shooting for end of Saturday.

Should I only work in v2?

**Comment by Ishakaru** [comment]
Start of saterday update: Things are progressing. Getting this to work feels messy. At least the using it won't be. So far I have:The button that closes the settings menu when opening the auto menu (working on a way to do the reverse as well). The popup from the bottom that can receive items like the settings menu. A way to create CSS to mold what's there to what we need.

Edit: got the settings/auto interaction solved. Working on a way to make the code interface usable.

**Comment by Ishakaru** [comment]
Sunday: Issues with things not working like I think they should. Fixed. Toggle buttons done. Working on value inputs.

**Comment by spindrjr** [comment]
Let me know if you need help with anything on this.

**Comment by Ishakaru** [comment]
I think I'm close. It's the whole script injection part that makes this so difficult. I'm out of time, I need to hit the sack. 
https://github.com/Ishakaru/AutoTrimps/blob/master/UI%20work

copa pasta into what ever version you're using. and it should work it's magic... all cept the freaking accept button... can NOT figure out what the issue is. There are plenty of comments, but if you have any questions, let me know.

**Comment by spindrjr** [comment]
Hey man this looks really good and it does look like you are very close. First of all let me say that my abilities to work with this injection stuff is limited to pretty much imitation. I can see how you made a button and called the function to make it and I can do that again to make another button. That being said I won't be much help in working out bugs or adding features to stuff like this.

If the custom number entry thing is causing you undue trouble, maybe try cutting the suffix translation? Beyond that I know you don't want text boxes sitting there but personally I have no problem with it. If text boxes are simple and the custom number thing is being frustrating, maybe just go with text boxes for now to get it off the ground.

Anyway just my 2 cents. Thanks for the great work so far it looks fantastic. Maybe Zin or someone else will be more help in figuring out the bug.

**Comment by Ishakaru** [comment]
Yea, I may just go with text-boxes... but that kinda poses another issue. But we'll burn that bridge when we get to it.

This much I know: No code in autoSetValue() is getting run. So the suffix stuff (ripped from trimps btw) is not the issue.

There was something that kept nagging at me, hence why I'm up rather than asleep right now. I did some getto-debugging (console log >.<)  and here's the deal: 'George' is my test case. Notice the cap "G" and no space. The console log returned the correct string. The apply button onclick attribute, using the exact same variable I used for the log is written " george" in the html. I can see no rhyme or reason for this. As a troubleshooting test, I commented out all code in autoSetValue() and set it to toolTipUnlock(); and tooltip('hide');  which gave the exact same issue... This on top of the whole system throwing a fit from the get go when I change onclick to tooltip('hide').

Something smells funny, and I'm not talking about the weed. =-P Why would it work(badly, with random changes) one way, and completely break on something that's used all the time everywhere else?

edit: took one last look... think there might be an issue with the quotes somehow. I'll mess with it tomorrow.

edit2: couldn't sleep. This kept bugging the snot out of me and I couldn't leave it alone. I wanted to do it the right way.... but after a couple hours of tossing and turning, and going back to the comp to try something newish... I gave up.  1 random global string later: TADAH! got the value input done. I put the updated version up at https://github.com/Ishakaru/AutoTrimps/blob/master/UI%20work

What's left to do: Formatting, tables to put things in order. A few new CSS classes(function to do that included) to handle looks and what not. Formatting the code to be a little less hideous.
 A note on usage. All checks for data (toggled on or not or value) should be pulled from automationSettings with the appropriate index. Load should be done first(and making sure the array is the correct size). Default value before a load cycle happens is 0.

**Comment by spindrjr** [comment]
@Ishakaru , I messed around with this for a few hours. I think it works but I got stuck trying to make it more usable with the setting names. Correct me if I'm wrong, but it seems like in order to reference all of these settings, we need to use an arbitrary index number? Would it be possible to instead use an array of objects so that we can reference settings by their name?

Something like was done here with autoTSettings:
http://demoorken.github.io/autotrimpNOPOTENCY/autotrimp.js

I tried to do it by myself but of course it didn't work. The script above seems to do all of his functions outside of the html injection, which seems much easier. Do we have to inject all of ours just to use the bottom settings menu as opposed to the above script's popup menu?

**Comment by Ishakaru** [comment]
The injection is fire and forget. Shouldn't need to touch it at all. And yes it's needed to have interactive buttons on the page.... as well as the array. I took a look at that script, and it injects everything into the page.(look at the readme)

The array+object array was used because I'm a C++ programmer not javascript/html. =-P  From the look of it, I would need to create another function that would need to be injected to do what you want. I don't know how to pull it off with the flexibility I'm wanting to provide, but I'll see what I can do.

No, no extra injection script. I have an idea of how to make it more friendly. Just need to refoctor some stuff... need to head to work >.>

**Comment by spindrjr** [comment]
Ish and Zin have v2 interface up and running. Looks dope.
